server/boards/app/files.go

```diff
@@ -90,38 +69,62 @@ func (a *App) GetFileInfo(filename string) (*mm_model.FileInfo, error) {
 }
 
 func (a *App) GetFile(teamID, rootID, fileName string) (*mm_model.FileInfo, filestore.ReadCloseSeeker, error) {
-	fileInfo, err := a.GetFileInfo(fileName)
-	if err != nil && !model.IsErrNotFound(err) {
-		a.logger.Error("111")
+	fileInfo, filePath, err := a.GetFilePath(teamID, rootID, fileName)
+	if err != nil {
+		a.logger.Error("GetFile: Failed to GetFilePath.", mlog.String("Team", teamID), mlog.String("board", rootID), mlog.String("filename", fileName), mlog.Err(err))
 		return nil, nil, err
 	}
 
-	var filePath string
-
-	if fileInfo != nil && fileInfo.Path != "" {
-		filePath = fileInfo.Path
-	} else {
-		filePath = filepath.Join(teamID, rootID, fileName)
-	}
-
 	exists, err := a.filesBackend.FileExists(filePath)
 	if err != nil {
-		a.logger.Error(fmt.Sprintf("GetFile: Failed to check if file exists as path. Path: %s, error: %e", filePath, err))
+		a.logger.Error("GetFile: Failed to check if file exists as path. ", mlog.String("Path", filePath), mlog.Err(err))
 		return nil, nil, err
 	}
-
 	if !exists {
 		return nil, nil, ErrFileNotFound
 	}
 
 	reader, err := a.filesBackend.Reader(filePath)
 	if err != nil {
-		a.logger.Error(fmt.Sprintf("GetFile: Failed to get file reader of existing file at path: %s, error: %e", filePath, err))
+		a.logger.Error("GetFile: Failed to get file reader of existing file at path", mlog.String("Path", filePath), mlog.Err(err))
 		return nil, nil, err
 	}
 	return fileInfo, reader, nil
 }
 
+func (a *App) GetFilePath(teamID, rootID, fileName string) (*mm_model.FileInfo, string, error) {
+	fileInfo, err := a.GetFileInfo(fileName)
+	if err != nil && !model.IsErrNotFound(err) {
+		return nil, "", err
+	}
+
+	var filePath string
+
+	if fileInfo != nil && fileInfo.Path != "" {
+		filePath = fileInfo.Path
+	} else {
+		filePath = filepath.Join(teamID, rootID, fileName)
+	}
+
+	return fileInfo, filePath, nil
+}
+
+func getDestinationFilePath(isTemplate bool, teamID, boardID, filename string) string {
```

Comment:
Should we replace other calls to `filepath.Join(teamID, rootID, fileName)` with this?

Reply:
I don't think so. I would be artificially passing in true for `isTemplate`. In order to for getDestinationFilePath() to work in the proper manner. All of the remaining calls to `filepath.Join()` are dealing with only the OLD storage method, so they will never want to return the BaseFilePath.